From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 15 Mar 2018 19:02:28 +0000
Subject: i40e: Add kconfig dependency to ensure cmpxchg64() is available

i40e again uses cmpxchg64(), which isn't available in some 32-bit SMP
configurations.  It seems reasonable to expect that 40G hardware is
not used on 32-bit systems.

Fixes: 60f481b97038 ("i40e: change flags to use 64 bits")
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 drivers/net/ethernet/intel/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/intel/Kconfig b/drivers/net/ethernet/intel/Kconfig
index 1feb54b6d92e..dd17bdf5561e 100644
--- a/drivers/net/ethernet/intel/Kconfig
+++ b/drivers/net/ethernet/intel/Kconfig
@@ -215,6 +215,7 @@ config I40E
 	tristate "Intel(R) Ethernet Controller XL710 Family support"
 	imply PTP_1588_CLOCK
 	depends on PCI
+	depends on 64BIT || X86_CMPXCHG64
 	---help---
 	  This driver supports Intel(R) Ethernet Controller XL710 Family of
 	  devices.  For more information on how to identify your adapter, go
